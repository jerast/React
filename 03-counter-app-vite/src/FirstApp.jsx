import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle, name }) => {
	return (
		<>
			<h1 data-testid="test-title"> {title} </h1>
			<p>{subtitle}</p>
			<p>{subtitle}</p>
			<p>{name}</p>
		</>
	);
};

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subtitle: PropTypes.string,
};

FirstApp.defaultProps = {
	// title: 'User no identified',
	subtitle: 'There is no subtitle',
	name: 'Jose Rodriguez!!',
};
