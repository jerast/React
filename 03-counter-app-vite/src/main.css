html, body {
   background-color: #223;
   color: white;
   font-family: Arial, Helvetica, sans-serif;
   font-size: 1.3rem;
   padding: 70px;
   text-align: center;
   user-select: none;
}

p {
   font-size: 0.8rem
}

p > span {
   font-weight: 700;
}

button {
   border-radius: 16px;
   border: none;
   cursor: pointer;
   padding: 8px 24px;
   margin: 4px;
}
