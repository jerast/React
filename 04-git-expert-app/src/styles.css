* {
	font-family: Helvetica, Arial, sans-serif;
	background-color: #eaeaea;
}

body {
	padding: 60px;
}

input {
	background-color: white;
	border-radius: 5px;
	border: 1px solid #61209e;
	color: #000000;
	font-size: 1.2rem;
	outline: none;
	padding: 10px 15px;
	width: 100%;
	box-sizing: border-box;
}

h2 {
	font-size: 1.5rem;
}

h3 {
	font-size: 3rem;
	margin-bottom: 5px;
}

.card-grid {
	display: grid;
	grid-gap: 20px;
	justify-items: center;
	grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
	grid-template-rows: masonry;
}

.card {
	align-content: center;
	align-items: center;
	background-color: white;
	border-radius: 10px;
	box-shadow: 0px 3px 5px #00000008;
	display: flex;
	width: 100%;
	height: fit-content;
	flex-direction: column;
	justify-content: center;
	overflow: hidden;
}

.card p {
	background-color: #ffffff;
	flex: 1;
	font-size: 0.7rem;
	margin-top: 5px;
	padding: 5px 20px 0px 20px;
	text-align: center;
}

.card img {
	width: 100%;
}
