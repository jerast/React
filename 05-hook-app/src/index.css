@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
	* {
		@apply box-border;
	}

	body {
		@apply bg-zinc-800;
	}
	#root {
		@apply w-full grid gap-3 justify-items-center p-5  text-white;
	}

	.title {
		@apply text-center text-4xl my-4;
	}
	.subtitle {
		@apply text-center text-2xl my-4;
	}

	button {
		@apply px-5 py-2 bg-zinc-500 text-white rounded-full disabled:opacity-50 disabled:cursor-default;
	}
	input {
		@apply bg-zinc-500 px-5 py-2 rounded-xl w-full;
	}

	blockquote {
		@apply w-full max-w-[500px];
	}
	blockquote p {
		@apply font-[cursive] text-6xl m-3 text-center;
	}
	blockquote span {
		@apply text-end w-full inline-block font-light font-mono italic text-sm;
	}
}
