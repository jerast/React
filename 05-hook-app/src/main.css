@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
	* {
		@apply box-border select-none;
	}

	body {
		@apply bg-zinc-800;
	}
	#root {
		@apply w-full grid gap-3 justify-items-center p-5  text-white;
	}

	.title {
		@apply text-center text-4xl;
	}
	.subtitle {
		@apply text-center text-xl text-zinc-400;
	}

	button {
		@apply px-5 py-2 bg-zinc-700 text-white rounded-full disabled:opacity-50 disabled:cursor-default outline-none ease-in-out duration-75 hover:bg-zinc-800;
	}
	input {
		@apply bg-zinc-600 px-5 py-2 rounded-xl w-full;
	}
	form {
		@apply flex rounded-lg overflow-hidden shadow-md my-8
	}
	form > * {
		@apply rounded-none outline-none 
	}

	ul {
		@apply grid gap-2 w-full max-w-[500px]
	}
	.list-item {
		@apply w-full relative h-12 cursor-pointer ease-in-out duration-75
	}
	.list-item span {
		@apply w-full h-full rounded-full inline-flex items-center bg-zinc-500  hover:bg-zinc-600 ease-in-out duration-75 pl-5 shadow-md;
	}
	.list-item button {
		@apply text-2xl absolute right-0 px-5
	}
	.list-item.done {
		@apply  opacity-30;
	}
	.list-item.done span {
		@apply bg-zinc-600 line-through shadow-none;
	}

	blockquote {
		@apply w-full max-w-[500px];
	}
	blockquote p {
		@apply font-[cursive] text-6xl m-3 text-center;
	}
	blockquote span {
		@apply text-end w-full inline-block font-light font-mono italic text-sm;
	}
}
