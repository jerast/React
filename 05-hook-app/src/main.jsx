import React from 'react';
import ReactDOM from 'react-dom/client';
import { Padre } from './07-tarea-memo/Padre';
import './index.css';

// import { HooksApp } from './HooksApp'
// import { CounterApp } from './01-State/CounterApp'
// import { CounterWithCustomHook } from './01-State/CounterWithCustomHook';
// import { FormWithCustomHook } from './02-Effect/FormWithCustomHook';
// import { SimpleForm } from './02-Effect/SimpleForm';
// import { MultipleCustomHooks } from './03-Examples/MultipleCustomHooks';
// import { FocusScreen } from './04-Ref/FocusScreen';
// import { Layout } from './05-LayourEffect/Layout';
// import { Memorize } from './06-Memos/Memorize';
// import { MemoHook } from './06-Memos/MemoHook';
// import { CallbackHook } from './06-Memos/CallbackHook';

ReactDOM.createRoot(document.getElementById('root')).render(
	<>
		{/* <React.StrictMode> */}
			{/* <HooksApp /> */}
			{/* <CounterApp /> */}
			{/* <CounterWithCustomHook /> */}
			{/* <SimpleForm /> */}
			{/* <FormWithCustomHook /> */}
			{/* <MultipleCustomHooks /> */}
			{/* <FocusScreen /> */}
			{/* <Layout /> */}
			{/* <Memorize /> */}
			{/* <MemoHook /> */}
			{/* <CallbackHook /> */}
			<Padre />
		{/* </React.StrictMode> */}
	</>
);
