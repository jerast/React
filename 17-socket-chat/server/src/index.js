/**
 * @file index.js
 * @description Main entry point of the application.
 */

import { Server } from './connection/server.js'

/**
 * Initializes and runs the server.
 */
new Server().run()
