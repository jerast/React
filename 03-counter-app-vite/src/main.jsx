import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import './main.css';

ReactDOM.createRoot(document.getElementById('root')).render(
	<React.StrictMode>
		{/* <FirstApp title='Hola, soy Jerast'/> */}
		<CounterApp value={0}/>
	</React.StrictMode>
);
  