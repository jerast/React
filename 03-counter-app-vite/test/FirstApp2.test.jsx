import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FisrtApp />', () => {
	
	const title = 'Hola, soy Jerast';
	const subtitle = 'Soy un subtitulo';

	test('Debe de hacer match con el snapshot', () => {
		const { container } = render( <FirstApp title={ title } /> );
		expect( container ).toMatchSnapshot();
	});

	test('Debe de mostrar el mensaje "Hola, soy Jerast"', () => {
		render( <FirstApp title={ title } /> );
		expect( screen.getByText(title) ).toBeTruthy();
	});

	test('Debe mostrar el tÃ­tulo en un h1', () => {
		render( <FirstApp title={ title } /> );
		expect( screen.getByRole('heading', { level: 1 }).textContent ).toContain( title );
	});

	test('Debe de mostrar el subtitlulo envaido por props', () => {
		render( <FirstApp title={ title } subtitle={ subtitle } /> );
		expect( screen.getAllByText(subtitle).length ).toBe( 2 );
	});

});
