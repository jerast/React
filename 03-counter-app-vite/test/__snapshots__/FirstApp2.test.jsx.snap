// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FisrtApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Jerast
     
  </h1>
  <p>
    There is no subtitle
  </p>
  <p>
    There is no subtitle
  </p>
  <p>
    Jose Rodriguez!!
  </p>
</div>
`;
